### PR TITLE
feat(dpp): optional execution context in rs-dpp

### PR DESCRIPTION
**Pull request by qrayven**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/src/data_contract/data_contract.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use dpp::util::string_encoding::Encoding;
 use crate::errors::{from_dpp_err, RustConversionError};
 use crate::identifier::identifier_from_js_value;
 use crate::metadata::MetadataWasm;
-use crate::utils::WithJsError;
+use crate::utils::{Inner, WithJsError};
```
Comment (antouhou) [inline packages/wasm-dpp/src/data_contract/data_contract.rs:16]: Clippy is a bit unhappy here :)

File: packages/wasm-dpp/src/state_repository.rs [hunk 2]
```diff
@@ -723,3 +726,12 @@ impl StateRepositoryLike for ExternalStateRepositoryLikeWrapper {
         bail!("fetching latest platform block failed: value is not number");
     }
 }
+
+fn ctx_to_js_value(maybe_ctx: Option<&StateTransitionExecutionContext>) -> JsValue {
+    if let Some(ctx) = maybe_ctx {
+        let ctx_wasm: StateTransitionExecutionContextWasm = ctx.into();
+        ctx_wasm.into()
+    } else {
+        JsValue::null()
```
Comment (antouhou) [inline packages/wasm-dpp/src/state_repository.rs:735]: I think it's better to use `JsValue::undefined` here, as it's more conventional for "none" values in js

File: packages/wasm-dpp/src/data_contract/data_contract.rs (hunk 1 above)
Comment (qrayven) [reply packages/wasm-dpp/src/data_contract/data_contract.rs:16]: fixed

File: packages/wasm-dpp/src/state_repository.rs (hunk 2 above)
Comment (qrayven) [reply packages/wasm-dpp/src/state_repository.rs:735]: thanks, I was wondering what I should put here.

**Review by antouhou — APPROVED** [review]

**Pull request merged**